### DISCUSSION TITLE
Enable GEMMT from OpenBLAS

### DISCUSSION THREAD

**Discussion by jf-huang** [scivision/mumps-superbuild#43] [Q&A]
Hi everyone, 

I was trying to use the GEMMT functions available in the OpenBLAS package ([here](https://github.com/xianyi/OpenBLAS/issues/861) and [here](https://github.com/xianyi/OpenBLAS/pull/3548)). I compiled the latest OpenBLAS on my machine (ubuntu 20.04) but had no clue how to change the cmake files so that mumps can use it. Actually I am open to other BLAS packages as well if they have GEMMT but OpenBLAS seems to be a convenient choice. Any ideas?

Thanks for your input!

**Comment by scivision** [comment]
the MUMPS/cmake/lapack.cmake automatically tests for the presence of GEMMT. By default in cmake/options.cmake gemmt is "on" if available with MUMPS >= 5.2. Is MUMPS CMake saying that like "BLAS_HAVE_dGEMMT" is not working?

**Reply by jf-huang** [reply]
Hi @scivision , the part of the cmake file for detecting GEMMT in lapack is working good. My issue is that the BLAS and OpenBLAS versions installed using "apt install" (ubuntu 22.04) seem to not have GEMMT so cmake shows "Performing Test BLAS_HAVE_dGEMMT - Failed" and has GEMMT disabled during compilation. I then manually compiled the latest OpenBLAS (which is claimed to have GEMMT) and have it installed in the default location (/opt/OpenBLAS). However, cmake was not able to find it and I do not know how to change the cmake files to let it do so. The output is shown below, where the old BLAS in /usr/lib/x86_64-linux-gnu/ is still used:
```
-- MUMPS 5.5.1.6  CMake 3.22.1  Toolchain    
-- Performing Test LAPACK_s_FOUND            
-- Performing Test LAPACK_s_FOUND - Success  
-- Performing Test LAPACK_d_FOUND            
-- Performing Test LAPACK_d_FOUND - Success  
-- Found LAPACK: /usr/lib/x86_64-linux-gnu/liblapack.so;/usr/lib/x86_64-linux-gnu/libblas.so  found components: Netlib         
-- Performing Test BLAS_HAVE_dGEMMT          
-- Performing Test BLAS_HAVE_dGEMMT - Failed 
-- Found MPI_C: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so (found version "3.1") 
-- Found MPI_Fortran: /usr/lib/x86_64-linux-gnu/libmpi_usempif08.so (found version "3.1")                                      
-- Found MPI: TRUE (found version "3.1") found components: C Fortran                  
-- Performing Test SCALAPACK_d_FOUND         
-- Performing Test SCALAPACK_d_FOUND - Success                                        
-- Performing Test SCALAPACK_s_FOUND         
-- Performing Test SCALAPACK_s_FOUND - Success
```

**Reply by scivision** [reply]
specify the location to your own OpenBLAS with "LAPACK_ROOT" like

```sh
cmake -Bbuild -DLAPACK_ROOT=/opt/OpenBLAS
```
